product.js

```diff
@@ -0,0 +1,75 @@
+let colors = ['red', 'white', 'black', 'pink', 'yellow', 'green', 'blue', 'turqoise', 'custom'];
+let engines = ['engine_1', 'engine_2', 'engine_3', 'engine_4', 'engine_5', 'enigine_6', 'engine_7'];
+
+
+function Product() {
+    this.color = colors[Math.floor(Math.random() * colors.length)];
+    this.engine = engines[Math.floor(Math.random() * engines.length)];
+    this.isDriven = false;
+    this.mileage = 0;
+}
+
+
+
+function AssemblyLine(capacity = 200) {
+    this.capacity = capacity;
+    this.products = [];
+}
+
+AssemblyLine.prototype.generateProducts = function () {
+
+    if (this.capacity == 0) {
+        console.log("Sorry, the assembly line is not in use at the moment..");
+        return;
+    }
+
+    var index;
```

Comment:
Nu este nevoie sa declari index-ul aici, il poti pune direct in functia for, iar in loc de var, poti folosi let
`for( let i = 0; i < length; i++ )`